src/FunctionApps/DSTP/src/main/kotlin/gov/cdc/prime/phdi/Utilities/HL7Validator.kt

```diff
@@ -0,0 +1,26 @@
+package gov.cdc.prime.phdi.Utilities
+
+import ca.uhn.hl7v2.DefaultHapiContext
+import ca.uhn.hl7v2.HL7Exception
+import ca.uhn.hl7v2.HapiContext
+import ca.uhn.hl7v2.parser.Parser
+import ca.uhn.hl7v2.validation.ValidationContext
+import ca.uhn.hl7v2.validation.impl.ValidationContextFactory
+import ca.uhn.hl7v2.Version
+import ca.uhn.hl7v2.validation.builder.support.DefaultValidationBuilder
+
+public class HL7Validator() {
+    public fun isValidHL7Message(message: String): Boolean {
+        val context = DefaultHapiContext()
+        // We will likely replace this with a more custom validation class in the future
+        context.setValidationContext(ValidationContextFactory.defaultValidation() as ValidationContext);
+        val parser = context.getPipeParser()
+
+        try {
+            parser.parse(message)
+            return true
+        } catch (e: HL7Exception) {
+            return false
```

Comment:
Should there be logging here ? It looks to me like this would fail silently ?

Reply:
For this very basic approach that we're using right now, I think failing silently is okay. The gist of this snippet is to try to parse the HL7 message using the default validator, and if it fails for any reason related to HL7 then the message is deemed invalid. This does raise two points: 1) do we want to log _why_ the message failed and use that to inform the sender, and 2) do we need to incorporate a second `catch` condition that grabs all other exceptions (those not related to HL7)?

Reply:
if there's some way to do #1 that does not expose sensitive data, yes pls. and #2 yes but probably one level up from this function?

Reply:
I'm going to add logging and error in the next iteration of this code as the process is going to change slightly. I'll address these concerns in that PR. Resolving this comment for now.